Mathlib/Algebra/Lie/Derivation/Basic.lean

```diff
@@ -120,6 +120,38 @@ theorem ext_of_lieSpan_eq_top (s : Set L) (hs : LieSubalgebra.lieSpan R L s = 
     (h : Set.EqOn D1 D2 s) : D1 = D2 :=
   ext fun _ => eqOn_lieSpan h <| hs.symm ▸ trivial
 
+/-- The general Leibniz rule for Lie derivatives. -/
+theorem pow_leibniz (D : LieDerivation R L L) (n : ℕ) (a b : L) :
+    D.toLinearMap^[n] ⁅a, b⁆ = ∑ i in Finset.range (n + 1),
+      Nat.choose n i • ⁅D.toLinearMap^[i] a, D.toLinearMap^[n - i] b⁆ := by
```

Comment:
```suggestion
theorem iterate_apply_lie (D : LieDerivation R L L) (n : ℕ) (a b : L) :
    D^[n] ⁅a, b⁆ = ∑ i in Finset.range (n + 1), Nat.choose n i • ⁅D^[i] a, D^[n - i] b⁆ := by
```

Reply:
Brilliant! Yes, I much prefer using the antidiagonal, thanks for the suggestion!

Reply:
It's so much better now! I decided to keep the previous version also, since it seems one has to explicitly provide the function inside the sum when calling `Nat.sum_antidiagonal_eq_sum_range_succ` which can be annoying.